needle-slack-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,314 @@
+import type {
+  ConnectorRequest,
+  CreateConnectorRequest,
+} from "~/models/connectors-models";
+import {
+  createConnector,
+  getConnector,
+  listConnectors,
+  deleteConnector,
+  publishConnectorRun,
+  type Session,
+  type ConnectorRunDescriptor,
+  createNeedleFileId,
+} from "@needle-ai/needle-sdk";
+import { type CanvasFileMetadata, slackConnectorsTable } from "../db/schema";
+import { db } from "../db";
+import { eq } from "drizzle-orm";
+import {
+  getConnectorDetails,
+  getCurrentFiles,
+  handleDatabaseUpdates,
+  updateLastSynced,
+  createSlackConnectorRecord,
+  deleteSlackConnectorRecords,
+  getSlackConnectorWithFiles,
+} from "./slack-db";
+import {
+  computeCanvasDiff,
+  createNewFiles,
+  processExistingFiles,
+} from "./dif-utils";
+import { createSlackService } from "../slack/service";
+import { type ExistingFile } from "../slack/types";
+
+export async function createSlackConnector(
+  params: CreateConnectorRequest,
+  session: Session,
+) {
+  if (!params.channels.length) {
+    throw new Error("At least one channel must be provided");
+  }
+
+  const connector = await createConnector(
+    {
+      name: params.name,
+      cronJob: params.cronJob,
+      cronJobTimezone: params.cronJobTimezone,
+      collectionIds: [params.collectionId],
+      credentials: params.credentials,
+    },
+    session.id,
+  );
+
+  try {
+    await createSlackConnectorRecord(
+      connector.id,
+      params.channels,
+      params.timezone,
+    );
+    await runSlackConnector({ connectorId: connector.id }, session);
+    return connector;
+  } catch (error) {
+    await deleteConnector(connector.id, session.id);
```

Comment:
This `catch` block seems very unlikely - in those cases i'm ok not handling errors.

Reply:
+ if `runConnector` throws we delete the connector. which is not intuitive behaviour from user's perspective.

Reply:
Yes agreed. I've removed the try-catch block. Also fixes the unintuitive behavior where we were deleting the connector if runSlackConnector failed.